gnome-initial-setup/gis-driver.c

```diff
@@ -127,14 +132,67 @@ check_for_live_boot (gchar **uuid)
   return live_boot;
 }
 
+static void
+udisks_client_new_cb (
+    GObject *source,
+    GAsyncResult *result,
+    gpointer user_data)
+{
+  GisDriver *driver = GIS_DRIVER (user_data);
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  UDisksClient *udisks = NULL;
+  GError *error = NULL;
+
+  udisks = udisks_client_new_finish (result, &error);
+  if (udisks != NULL)
+    {
+      GList *blocks = udisks_client_get_block_for_uuid (udisks, priv->live_mode_uuid);
+      GList *l;
+
+      /* If you have the same ISO on a USB and a DVD and you insert both, it is
+       * hard to determine which is booted using UDisks, so assume it's the DVD.
+       */
+      for (l = blocks; l != NULL && !priv->is_live_dvd; l = l->next)
+        {
+          UDisksBlock *block = UDISKS_BLOCK (l->data);
+          UDisksDrive *drive = udisks_client_get_drive_for_block (udisks, block);
+
+          if (drive != NULL && udisks_drive_get_optical (drive))
+            {
+              priv->is_live_dvd = TRUE;
+              g_object_notify_by_pspec (G_OBJECT (driver), obj_props[PROP_LIVE_DVD]);
```

Comment:
Asked another way: who would cancel the cancellable? Are you proposing that the `udisks_client_new()` call would not own a ref to the `GisDriver` and the driver would cancel an internal cancellable when disposed?